jax/_src/stages.py

```diff
@@ -299,6 +299,9 @@ class ArgInfo:
   _aval: core.AbstractValue
   donated: bool
 
+  def __len__(self):
+    return self._aval.__len__()
```

Comment:
I don't think `__len__` exists on AbstractValue: `AttributeError: 'ShapedArray' object has no attribute '__len__'`

So maybe we shouldn't add it to ArgInfo also?